js/main.js

```diff
@@ -1,2 +1,43 @@
 /* global data */
 /* exported data */
+
+var $form = document.querySelector('form');
+var $image = document.querySelector('img');
+var $photo = document.querySelector('#photo');
+var $title = document.querySelector('#title');
+var $notes = document.querySelector('#notes');
+
+function handleInputEvent(event) {
+  if (event.target !== $photo) {
+    return;
+  }
+  $image.setAttribute('src', $photo.value);
+}
+
+function handleSubmitEvent(event) {
+  if (event.target !== $form) {
+    return;
+  }
+  var form = {
+    title: $title.value,
+    photoUrl: $photo.value,
+    notes: $notes.value
+  };
+
+  var previousDataJSON = JSON.parse(localStorage.getItem('form-user-data'));
+
+  if (previousDataJSON === null) {
```

Comment:
All localStorage-related code should be in `data.js`, in the same file with the variable `data` that is being preserved in localStorage